chroot_setup

```diff
@@ -0,0 +1,428 @@
+def chroot_system(line: str):
+  line = line.rstrip()
+  print(f"system -i {line}")
+  if not CHROOT_DEBUG:
+    os.system(f"system -i {line}")
+
+# :sh
+# Executes a line from a .lst file using Pythons os.system function, which is
+# run "by calling the Standard C function system()."
+def chroot_sh(line: str):
+  print(line)
+  if not CHROOT_DEBUG:
+    os.system(line)
+
+# :file
+# Parses and executes additional .lst files
+def chroot_setup (chroot_lst):
+  lst_file = open(chroot_lst, 'r')
+  action = None
+
+  # Iterate over all lines in the file, executing actions where encountered
+  for line in lst_file:
+    line = line.strip()
+
+    # Skip blank lines and comments
+    if not line or line.startswith('#'):
+      continue
+
+    # Determine action
+    elif line.startswith(':'):
+      action = actions.get(line, None)
+
+    # Execute action
+    else:
+      if action is not None:
+        action(line)
+
+# Mapping the .lst action keys to the functions that run on their values
+actions = {
+  ':file'       : chroot_setup,
+  ':mkdir'      : chroot_mkdir,      # done
+  ':ln_fix_rel' : chroot_ln_fix_rel, # done
+  ':ln_rel'     : chroot_ln_rel,     # done
+  ':ln'         : chroot_ln,         # done
+  ':mknod'      : chroot_mknod,      # done
+  ':cp_dir'     : chroot_cp_dir,     # done
+  ':cp'         : chroot_cp,         # done
+  ':chmod_dir'  : chroot_chmod_dir,  # done
+  ':chmod'      : chroot_chmod,      # done
+  ':chown_dir'  : chroot_chown_dir,  # done
+  ':chown'      : chroot_chown,      # done
+  ':system'     : chroot_system,     # done
+  ':sh'         : chroot_sh          # done
+}
+
+# Run an initial check on the environment, ensuring that we are running on IBM i
+# and setting environment variables
+def initial_check():
+  if (os.uname().sysname == 'OS400'):
+    # CHROOT_DEBUG = False
+    print("**********************")
+    print("Live IBM i session (changes made).")
+    print("**********************")
+
+  else:
+    # CHROOT_DEBUG = True
+    print("**********************")
+    print("Not IBM i, no action is taken (debug flow purpose only).")
+    print("**********************")
+    # exit -1
+
+
+# Install any yum packages passed with the -i flag
+def yum_install(package_list: list):
+  install_template = Template('/QOpenSys/pkgs/bin/yum $yes --installroot=$chroot_dir install $package')
+  auto_yes = '-y' if args.auto_yes else ''
+  for package in package_list:
+    command = install_template.substitute(yes=auto_yes, chroot_dir=args.chroot_directory, package=package)
+    logging.debug(command)
+    os.system(command)
+
+#
+# MAIN
+#
+if __name__ == '__main__':
+  parser = argparse.ArgumentParser()
+  # positional arguments
+  parser.add_argument("chroot_directory", metavar="CHROOT_DIRECTORY", help="The location of the chroot to be created")
+  parser.add_argument("chroot_type", metavar="CHROOT_TYPE", help="A list of .lst files to use for generating the chroot", nargs="?")
```

Comment:
chroot_type `nargs` should `*` (zero or more) instead of `?` (0 or 1)  because we can specify multiple lst files e.g. `minimal`,  `includes`, `nls`

Reply:
Fixed this up in d80220a